### ISSUE TITLE
Write short and clean readme

### ISSUE THREAD

**Issue by lgrapenthin** [Day8/re-frame#41]
I came here because in the clojurescript groups there lately is a lot of talk about re-frame.
I am sorry if this isn't what you want to be reading, but I attempted a second time to get an idea what your library does, so far without looking at the source code, and I believe that
1) Spending the entire evening or even longer reading a 1000 lines of README to understand what I am even looking at or am going to understand is not worth my or any ones time, even if your library implemented the theory of everything
2) Philosophical quotes and laissez-faire statements mixed with abstract concepts and insane paragraphs like "Correct acronym?", this one followed by a paragraph titled "Mostly a mashup" before any concrete code example or diagram aren't helping, either.
3) The paragraph "What problem does it solve" lacks a clear problem statement. 

I get that your library is about coordinating data flow in a web client written with reagent. Why don't you simply illustrate that with a few examples showing things in reagent (and preferably om as well) that you can't do without re-frame or that are at least typical to stumble upon, accompanied by a few better looking examples using re-frame? And put all the rest on a separate documentation page or into a wiki.

**Comment by mike-thompson-day8** [comment]
Thanks for the feedback.  It is disappointing to know that you find parts of the README insane and that it hasn't helped you to get any idea about the purpose of re-frame. 

I would have thought that the first few sections were pretty clear on what re-frame gives you:  **re-frame is an MVC-like  framework for writing SPAs using Reagent.**  

If you need one of those, then I'll think you'll find the rest of the README to be a tutorial.

**Comment by bendisposto** [comment]
First of all, I think the complaints are rude and not very constructive. Stop insulting and start improving, it is open source, you can send a pull request. I also have to disagree with the content. While there is of course room for improvement, the readme covered most of the things I wanted to know when starting to use re-frame.

**Comment by lgrapenthin** [comment]
Certainly it is a matter of opinion how a good README should look like. At the very least, my recommendations above are in line of what most READMES in the Clojure community look like. 

Documentation, long tutorials and extensive rationale are usually linked to. In my opinion, a README should lead with a clear problem statement, few motivating examples, and instructions how to get started on two screen pages at a maximum.

I don't think anything is wrong with the paragraphs mentioned above in general. They just don't do a very good job at introducing the library. Things like plays with acronyms or a reference to Apocalypse now surely are fun and motivating to read, for example to someone more familiar with how re-frame differs from the traditional MVC approach looking to consolidate his understanding and beliefs. But to someone new, those differences should be introduced first and by example. Otherwise it feels like observing people having fun without being invited to the party. Add statements like "But, for the most part, re-frame is just a mashup of emerging ideas." and he won't be very motivated to investigate further. 

I use om on a daily basis. Om has no such thing as re-frame. And there, too, control flow often imposes difficult challenges. If you make statements like "And I did actively reject the current ClojureScript fashion of using Cursors.", please state why or you have lost me. 

In a TLDR, this README does no good job at introducing and "selling" the library. It assumes that its reader is a fan already.

(@bendisposto If I was a found user of re-frame, I'd do exactly what you suggest: Improve and send in a PR. As stated clearly, I am not. Thus, raising this issue leaving my feedback is the most I could do. If I have had no constructive intentions, I'd simply have moved on without doing so.)

**Comment by gadfly361** [comment]
I don't think @lgrapenthin had any intention other than to offer advice and constructive criticism with valid concerns. 

With that said, it shouldn't be ignored that re-frame *exploded* onto the clojurescript and reagent scene specifically because of its untraditional README.  Sometimes it is simply nice to hear words of encouragement too, so I'd like to offer up the following:

https://twitter.com/kulesa/status/569924357205716992
https://twitter.com/toxi/status/571793506643152896
https://twitter.com/gokulj/status/576396290386362369

**Comment by bendisposto** [comment]
"is not worth my or any ones time, even if your library implemented the theory of everything" or "... mixed with abstract concepts and insane paragraphs  ..." is not an acceptable tone IMHO.

**Comment by danielytics** [comment]
I personally enjoyed reading it because of the style of README, but I do think it slowed me down a bit too and I had to come back and read it a second time.

I wouldn't change the style at all, but the introduction sections repeat themselves a lot before actually getting to the meat of what re-frame is and does in practical terms.

If it were my library, I would probably move the bulk of the current README into a wiki page and replace the README with a brief rationale (what it does and why we need it), getting started (eg clojars info), example, license. And prominently linking the "introductory guide" in the wiki (which would be the current README) as the next step to read.

Or maybe just rearrange the current README so that it has the basic "what it is, why we need it, how to use it version info and short example" is first and the current README follows.

Basically something to get people an at-a-glance view before diving into the full README, but I wouldn't want to dilute the content of the README itself (or at least leave it intact in the wiki) as it is a pretty good read and sets the tone for the library and rest of the documentation.

**Comment by krisleech** [comment]
I can offer a perspective of a someone is new to Clojure, ClojureScript, Reagent, ReactJS and ReFrame. I have read about functional programming over the years, dabbled, but this is the first time I have written production code in a FP lang. The only Javascript lib's I have used are JQuery, Underscore and Backbone.

I just finished reading the README. I think I got my head around ReFrame after detours to read various posts linked to in the README. For me, because many of the concepts are new, such as FRP, it was very helpful to have extra detail but I do think the README would benefit from being more concise. I like the chatty style of writing (and use of non-programming quotes!), but it is a little verbose in places. A wiki might be a better place to expand on each area. I'm not sure in the README we need to know anything about the internals, or more advanced topics like FSM, reduce - great stuff to read incidentally.

While I was reading the README I made notes for our team wiki (there isn't much out there tutorial wise on ReAgent), it might be helpful in seeing a newcomers perspective. Please excuse spelling mistakes. https://gist.github.com/krisleech/45717bac390a87ade4fe

**Comment by mike-thompson-day8** [comment]
I've made a couple of tweaks to the README so the initial part is more self contained (a slightly more a traditional README), and then more clearly separated it from the Tutorial which follows. 

But the changes are, admittedly, minor.

Other than that, I intend to keep the current conversational tone. I realise that some won't like it, seeing it as indulgent, but I personally prefer for some novelty to be sprinkled into the dry technical material I'm reading.

Also, Clojure has a rich history of this kind of attitude and who am I to break with tradition:
  - https://github.com/technomancy/leiningen
  - http://www.braveclojure.com/    (genius!)

In an unrelated change, I've added a "Where Do I Go Next?" section to the end of the Tutorial.
